Scuti/Scripts/UI/Store/TopBarView.cs

```diff
@@ -20,27 +20,42 @@ public class TopBarView : View {
     public List<BannerWidget> additionalBanners;
     private bool isAdditionalBanners;
 
+    
     private void Awake()
+    {
+        // Only the default banner is enrolled.        
+        Banner.onCreateBanners += CreateBanners;
+      
+    }
+
+    private void CreateBanners(int amount)
     {
         additionalBanners = new List<BannerWidget>();
 
+        Debug.Log("2. Topbar: total count: ---------------: "+amount);
+        if (amount < 2) return;
+
         widthBanner = rectBanner.sizeDelta.x + thresholdBanners;
         maxWidthContent = contentBanners.rect.size.x;
         int amountBanners = (int)(maxWidthContent / widthBanner);
+        Debug.Log("3. Topbar: amount banners: ---------------: " + amountBanners);
+        if (amountBanners > amount) amountBanners = amount;
 
-        if(amountBanners > 1)
+        if (amountBanners > 1)
         {
             // Instance banners
             for (int i = 0; i < amountBanners - 1; i++)
             {
                 BannerWidget banner = Instantiate(Banner, contentBanners.transform);
-                banner.gameObject.name = "Banner - "+ (int)(i+1);
-                banner.SetIndex(i+amountBanners);
+                banner.gameObject.name = "Banner - " + (int)(i + 1);
+                banner.SetIndex(i + amountBanners - 1);
```

Comment:
This needs to be capped under Amount. 

Lets say we have data from 5 banners.  We have 5 banners on screen. 

The fifth one would be (4+5-1) or 8. We have only 5 so this will be a blank spot on the screen. 

I think we need something like  floor(Total Banner Data Count / Total Visual Banners);

So if we have 5 data and 5 visuals then the increment is 1 (5/5)
If we have data for 100 banners and 5 visuals then the increment is 20 (100/5)
So it would be SetIndex(i*increment)

Important: Need to make sure increment is at least one. 
Important: Need to make sure the final value of SetIndex is below the number of banner data we have.

Reply:
That's right... i will change that